src/jit/morph.cpp

```diff
@@ -4440,113 +4690,134 @@ GenTreePtr    Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPt
                 }
             }
         }
-
-        // Check if we couldn't transform the LDOBJ(ADDR(LCLVAR)) into a struct promoted GT_LIST above
-        if (newArg == nullptr)
+        else
         {
             //
-            // We weren't able to pass this LclVar using it's struct promted fields
-            //
-            // So instead we will create a list of GT_LCL_FLDs nodes to pass this struct
+            // We will create a list of GT_LCL_FLDs nodes to pass this struct
             //
             lvaSetVarDoNotEnregister(varNum DEBUG_ARG(DNER_LocalField));
-
-            // If this is going in the register area, we transform it here into a GT_LIST of LCLFLD's
-            // If this is going in the outgoing arg area, it will be transformed later
-            //
-            if (fgEntryPtr->regNum != REG_STK)
-            {
-                GenTreeLclVarCommon* varNode = argValue->AsLclVarCommon();
-                unsigned   varNum = varNode->gtLclNum;
-                assert(varNum < lvaCount);
-                LclVarDsc* varDsc = &lvaTable[varNum];
-
-                GenTreePtr loLclFld = gtNewLclFldNode(varNum, type0, 0);
-                GenTreePtr hiLclFld = gtNewLclFldNode(varNum, type1, TARGET_POINTER_SIZE);
-
-                // Create a new tree for 'arg'
-                //    replace the existing LDOBJ(ADDR(LCLVAR)) 
-                //    with a LIST(LCLFLD-LO, LIST(LCLFLD-HI, nullptr))
-                //
-                newArg = gtNewListNode(loLclFld, gtNewArgList(hiLclFld));
-            }
         }
     }
-    // Check if we already created a replacement newArg above
+
+    // If we didn't set newarg to a new List Node tree
+    //
     if (newArg == nullptr)
     {
         if (fgEntryPtr->regNum == REG_STK)
```

Comment:
Say the param is a struct promoted and exactly has two fields of type long.  The above logic would create a GT_LIST of two struct promoted lcl vars.  Now we will have GT_PUTARG_STK (GT_LIST).  Is that handled by arm64 backend?  Wondering whether the above check should be even before line 4653